RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,108 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by sehong, kaki.
 //  Copyright © yagom academy. All rights reserved.
-// 
+//
+enum InputError: Error {
+    case invalidNumber
+}
+
+enum GameOptions {
+    case scissors
+    case rock
+    case paper
+    
+    init?(rawValue: Int) {
+          switch rawValue {
+          case 1:
+              self = .scissors
+          case 2:
+              self = .rock
+          case 3:
+              self = .paper
+          default:
+              return nil
+          }
+    }
+}
+
+enum GameResult {
+    case win
+    case lose
+    case draw
+    
+    var message: String {
+        switch self {
+        case .win:
+            return "이겼습니다!"
+        case .lose:
+            return "졌습니다!"
+        case .draw:
+            return "비겼습니다!"
+        }
+    }
+}
+
+enum GameMessage {
+    static let rockScissorsPaper = "가위(1), 바위(2), 보(3)! <종료 : 0> : "
+}
+
+func createComputerSelect() -> GameOptions {
+    guard let computerHand = GameOptions(rawValue: Int.random(in: 1...3)) else {
+        return createComputerSelect()
+    }
+    
+    return computerHand
+}
+
+func printMenu() {
+    print(GameMessage.rockScissorsPaper, terminator: "")
+}
+
+func receiveUserInput() throws -> Int {
+    guard let input = readLine(), let inputNumber = Int(input) else {
+        throw InputError.invalidNumber
+    }
+    
+    return inputNumber
+}
 
-import Foundation
+func judgeGameResult(comparing userHand: GameOptions, and computerHand: GameOptions) {
+    switch (userHand, computerHand) {
+    case (.scissors, .paper), (.rock, .scissors), (.paper, .rock):
+        print(GameResult.win.message)
+    case (.scissors, .rock), (.rock, .paper), (.paper, .scissors):
+        print(GameResult.lose.message)
+    case (.scissors, .scissors), (.rock, .rock), (.paper, .paper):
+        print(GameResult.draw.message)
+        play()
+    }
+}
 
-print("Hello, World!")
+func play() {
+    do {
+        printMenu()
+        let userInput = try receiveUserInput()
+        guard userInput != 0 else {
+            print("게임종료")
+            return
+        }
+        switch GameOptions(rawValue: userInput) {
+        case .scissors:
+            judgeGameResult(comparing: .scissors, and: createComputerSelect())
+        case .rock:
+            judgeGameResult(comparing: .rock, and: createComputerSelect())
+        case .paper:
+            judgeGameResult(comparing: .paper, and: createComputerSelect())
+        default:
+            throw InputError.invalidNumber
```

Comment:
```suggestion
        case .none:
            throw InputError.invalidNumber
```
문맥상 switch문을 이용하여 옵셔널을 바인딩 했을때에는 `default`보다 `case .none`이 더 자연스러운 것 같네요.
물론 지금도 괜찮습니다 참고만 해보시면 좋을 것 같아요!

Reply:
참고하도록 하겠습니다!